src/coprocessor/dag/storage/fixture.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::collections::{btree_map, BTreeMap};
+use std::sync::Arc;
+
+use super::range::*;
+use crate::coprocessor::{Error, Result};
+
+type ErrorBuilder = Box<dyn Send + Sync + Fn() -> Error>;
+
+type FixtureValue = std::result::Result<Vec<u8>, ErrorBuilder>;
+
+/// A `Storage` implementation that returns fixed source data (i.e. fixture). Useful in tests.
+#[derive(Clone)]
+pub struct FixtureStorage {
+    data: Arc<BTreeMap<Vec<u8>, FixtureValue>>,
+    data_view_unsafe: Option<btree_map::Range<'static, Vec<u8>, FixtureValue>>,
+    is_backward_scan: bool,
+    is_key_only: bool,
+}
+
+impl FixtureStorage {
+    pub fn new(data: BTreeMap<Vec<u8>, FixtureValue>) -> Self {
+        Self {
+            data: Arc::new(data),
+            data_view_unsafe: None,
+            is_backward_scan: false,
+            is_key_only: false,
+        }
+    }
+}
+
+impl From<Vec<(Vec<u8>, Vec<u8>)>> for FixtureStorage {
```

Comment:
Do we need to implement the `From<<Vec<(&[u8], &[8])>>>` for `FixtureStorage`? (I think it's necessary if elsewhere will construct a `FixtureStorage` from `From<<Vec<(&[u8], &[8])>>>` like: https://github.com/tikv/tikv/pull/5129/files#diff-8b90a8d482d4d26d1843a4c39a31868bR228)

Reply:
Good catch